src/libbson/src/bson/bson-iter.h

```diff
@@ -40,6 +40,15 @@ BSON_BEGIN_DECLS
 
 #define BSON_ITER_HOLDS_BINARY(iter) (bson_iter_type ((iter)) == BSON_TYPE_BINARY)
 
+#define BSON_ITER_HOLDS_VECTOR(iter) \
+   (BSON_ITER_HOLDS_BINARY (iter) && bson_iter_binary_subtype (iter) == BSON_SUBTYPE_VECTOR)
+
+#define BSON_ITER_HOLDS_VECTOR_INT8(iter) (bson_vector_int8_const_view_from_iter (NULL, iter))
+
+#define BSON_ITER_HOLDS_VECTOR_FLOAT32(iter) (bson_vector_float32_const_view_from_iter (NULL, iter))
+
+#define BSON_ITER_HOLDS_VECTOR_PACKED_BITS(iter) (bson_vector_packed_bits_const_view_from_iter (NULL, iter))
```

Comment:
```suggestion
#define BSON_ITER_HOLDS_VECTOR_PACKED_BIT(iter) (bson_vector_packed_bit_const_view_from_iter (NULL, iter))
```

The "BSON Binary Subtype 9" spec always refers to "PACKED_BIT" in the singular (i.e. "a PACKED_BIT vector"). Should we also be using the singular in our API?